### PR TITLE
fix: implement natural sorting for routeIds in /stop endpoint

### LINKED ISSUE #995: spec-gap: Implement natural sorting for routeIds in the /stop endpoint
### Description
While reviewing the `/stop` endpoint against the [wiki spec](https://github.com/OneBusAway/maglev/wiki/stop), I identified a gap in how `routeIds` are sorted in the response.

**Spec Requirement:** > "Ordered array... Sorted by route short name in natural string order (numeric digit runs are compared as numbers, so "14" < "101" < "B"). When short name is absent, long name is used for sorting."

**Current Behavior:**
The handler currently iterates over the routes exactly as they are returned by the database via `GetRoutesForStop`, which results in alphanumeric or insertion order, failing the natural sort requirement.

**Proposed Fix:**
1. Implement a `NaturalCompare` string utility in `internal/utils/`.
2. Use `slices.SortFunc` to sort the `routes` slice based on `ShortName` (falling back to `LongName`) before generating the `combinedRouteIDs` array in `stop_handler.go`.
3. Add a test in `stop_handler_test.go` to assert the "14" < "101" < "B" ordering.

### PR DESCRIPTION
**Pull request by ARCoder181105**
### Description

fixes #995 

This PR addresses a spec gap identified in the `/stop` endpoint. According to the [wiki spec](https://github.com/OneBusAway/maglev/wiki/stop), `routeIds` must be sorted in natural string order by `shortName` (falling back to `longName`), meaning `"14" < "101" < "B"`. Previously, the API returned routes in database fetch order. 

### Changes Made
- Added a `NaturalCompare` utility function to `internal/utils/` to handle string chunking and numeric comparisons.
- Updated `stopHandler` to use `slices.SortFunc` combined with `NaturalCompare` immediately after fetching the stop's routes from the DB.
- Added a targeted test `TestStopHandler_NaturalSorting` to explicitly verify the `"2" < "14" < "101" < "B"` ordering behavior.

### Testing
- [x] Added automated unit tests verifying natural sort.
- [x] Existing tests pass successfully.

image

## Summary by CodeRabbit

* **New Features**
  * Stop endpoint now returns routes in natural string order (numeric-aware), using short names when present and falling back to long names, improving route list readability.

* **Tests**
  * Added tests verifying natural sorting behavior, including numeric ordering, fallback name usage, and edge cases (leading zeros, large numbers).

### CODE REVIEW

**Comment by ARCoder181105** [comment]
@coderabbitai full review

**Review by burma-shave — COMMENTED** [review]
Reviewed findings from natural sort implementation.

File: internal/restapi/stop_handler_test.go [hunk 1]
```diff
@@ -203,3 +203,89 @@ func TestStopHandlerWithSituations(t *testing.T) {
...
+		name := route.ShortName
+		if name == "" {
+			name = route.LongName
+		}
+		returnedNames = append(returnedNames, name)
+	}
+
+	// Assert the array matches the spec's natural order
+	expectedOrder := []string{"2", "14", "101", "B", "Fallback"}
+	assert.Equal(t, expectedOrder, returnedNames, "Routes should be sorted in natural string order")
```
Comment (burma-shave) [inline internal/restapi/stop_handler_test.go]: This assertion checks sort order on `References.Routes`, but the spec doesn't require references arrays to be sorted — only `Entry.RouteIDs` and `Entry.StaticRouteIDs` have that requirement.

The assertion passes today because the handler appends routes directly from the sorted `routes` slice (not via a map), so order is preserved as a side effect. But if the route-reference builder is ever refactored to use a deduplication map (the canonical pattern elsewhere in this codebase), this test will fail intermittently even though the spec-required behavior is intact.

The `Entry.RouteIDs` assertion below already covers the spec requirement. Consider removing the `returnedNames` loop and this assertion.

**Review by burma-shave — CHANGES_REQUESTED** [review]

File: internal/restapi/stop_handler_test.go (hunk 1 above)
Comment (ARCoder181105) [reply internal/restapi/stop_handler_test.go]: Great catch, thanks @burma-shave ...
I've removed the `returnedNames` loop and that specific assertion entirely in the latest commit.

**Comment by ARCoder181105** [comment]
@coderabbitai full review

**Comment by ARCoder181105** [comment]
Addressed all the review comments....
PR ready for review...

**Review by burma-shave — APPROVED** [review]
